models/repo_tree.go

```diff
@@ -1,8 +1,79 @@
-// TODO: Create TreeListing and RepoFile and other model objects here, and then
```

Comment:
This file should be put in `github.com/go-gitea/git` not here.

Reply:
@lunny these types are for the API representation of the tree listing, matching the GitHub API, and to make the tree listing API easier to test -- otherwise they add little to the `git.Tree` and `git.TreeEntry` types that already exist. Would it be better to avoid defining public types, and instead add JSON annotations to those types in `go-gitea/git` ?

Reply:
If we're sending them over to the Client they should be defined in `go-gitea/go-sdk` 🙂